### PR TITLE
Create module-dependecies-tree

### PR DESCRIPTION
**Pull request by sergioboi**
move code that find all configuration ProjectDependencies there

### CODE REVIEW

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/BytecodeResolver.kt [hunk 1]
```diff
@@ -1,17 +0,0 @@
-package com.avito.impact
```
Comment (sergioboi) [inline subprojects/gradle/impact-shared/src/main/java/com/avito/impact/BytecodeResolver.kt:1]: unused

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/SimpleConfiguration.kt [hunk 2]
```diff
@@ -91,25 +92,16 @@ abstract class SimpleConfiguration(val module: InternalModule) : Equality {
             .toSet()
     }
 
+    open val dependencies: Set<ImplementationConfiguration> by lazy {
```
Comment (sergioboi) [inline subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/SimpleConfiguration.kt]: Now it's reused in each configuration

File: subprojects/gradle/module-dependencies-tree/src/main/kotlin/com/avito/module/configurations/ConfigurationType.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.avito.module.configurations
+
+import org.gradle.api.artifacts.Configuration
+import java.util.function.Predicate
+
+enum class ConfigurationType : Predicate<Configuration> {
```
Comment (sergioboi) [inline subprojects/gradle/module-dependencies-tree/src/main/kotlin/com/avito/module/configurations/ConfigurationType.kt:6]: Moved

File: subprojects/gradle/module-dependencies-tree/src/main/kotlin/com/avito/module/dependencies/ModuleConfigurationDependencies.kt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.avito.module.dependencies
+
+import com.avito.module.configurations.ConfigurationType
+import org.gradle.api.Project
+import org.gradle.api.internal.artifacts.dependencies.DefaultProjectDependency
+
+fun Project.dependenciesOnProjects(configurationType: ConfigurationType): Set<DefaultProjectDependency> {
```
Comment (sergioboi) [inline subprojects/gradle/module-dependencies-tree/src/main/kotlin/com/avito/module/dependencies/ModuleConfigurationDependencies.kt:7]: the first step for reusing code in impact and new module for demo-apps dependencies

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/SimpleConfiguration.kt [hunk 5]
```diff
@@ -91,25 +92,16 @@ abstract class SimpleConfiguration(val module: InternalModule) : Equality {
...
-                .dependencies
-                .withType(DefaultProjectDependency::class.java)
-        }
-        .toSet()
-        .map {
-            it.dependencyProject
-                .internalModule
-                .implementationConfiguration
-        }
-        .toMutableSet().apply {
```
Comment (sergioboi) [inline subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/SimpleConfiguration.kt:112]: @eugene-krivobokov I don't get what it is. I expect the existing unit tests for that scenario.
Could we discuss and write them?

Comment (sergioboi) [reply subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/SimpleConfiguration.kt:112]: Ok, I find that test

**Comment by eugene-krivobokov** [comment]
Didn't get the purpose of this change.
`impact-shared` still contains the key logic.

What is the responsibility of the new `module-dependencies-tree` in comparison to `impact-shared`?

**Comment by sergioboi** [comment]
> Didn't get the purpose of this change.
> `impact-shared` still contains the key logic.
> 
> What is the responsibility of the new `module-dependencies-tree` in comparison to `impact-shared`?

The code in `impact-shared` highly coupled and I don't know how to reuse logic about `impact` and `module dependencies`
So I extract a small part


The key differences I see that `impact-shared` must contain only `impact` logic without module dependency calculation

**Review by eugene-krivobokov — APPROVED** [review]

**Pull request merged**